pkg/provisioner/ironic/ironic.go

```diff
@@ -1613,6 +1617,33 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		_, err := p.hardPowerOff()
+		if err != nil {
+			result.Dirty = true
+			switch err.(type) {
+			case HostLockedError:
+				p.log.Info("could not power off host, busy")
+				return retryAfterDelay(powerRequeueDelay)
+			default:
+				// For other errors continue with the delete
+				if p.host.Status.ErrorCount > maxErrorRetryCount {
+					result.Dirty = false
+					p.host.Status.ErrorCount = 0
+					p.host.Status.ErrorMessage = ""
+				} else {
+					p.host.Status.ErrorCount++
+					p.host.Status.ErrorMessage = "failed to power off host"
+					return transientError(errors.Wrap(err, "failed to power off host"))
```

Comment:
`transientError` does not activate the retry mechanism with backoff, but `operationFailed()` is required instead. In addition to that, `BareMetalHostReconciler.actionDeleting` will require to be changed in order to manage properly the case when `provResult.ErrorMessage` is not empty so that `recordActionFailure` could be invoked.

Reply:
Done.